pkg/juicefs/mount/pod_mount.go

```diff
@@ -314,23 +313,38 @@ func (p *PodMount) JDeleteVolume(ctx context.Context, jfsSetting *jfsConfig.JfsS
 func (p *PodMount) genMountPodName(ctx context.Context, jfsSetting *jfsConfig.JfsSetting) (string, error) {
 	log := util.GenLog(ctx, p.log, "genMountPodName")
 	labelSelector := &metav1.LabelSelector{MatchLabels: map[string]string{
-		common.PodTypeKey:           common.PodTypeValue,
-		common.PodUniqueIdLabelKey:  jfsSetting.UniqueId,
-		common.PodJuiceHashLabelKey: jfsSetting.HashVal,
+		common.PodTypeKey:          common.PodTypeValue,
+		common.PodUniqueIdLabelKey: jfsSetting.UniqueId,
 	}}
 	pods, err := p.K8sClient.ListPod(ctx, jfsConfig.Namespace, labelSelector, nil)
 	if err != nil {
 		log.Error(err, "List pods of uniqueId", "uniqueId", jfsSetting.UniqueId, "hashVal", jfsSetting.HashVal)
 		return "", err
 	}
+	var podName string
 	for _, pod := range pods {
+		po := pod
+		if po.Labels[common.PodJuiceHashLabelKey] != jfsSetting.HashVal {
+			for k, v := range pod.Annotations {
+				if v == jfsSetting.TargetPath {
+					log.Info("Found pod with same target path, delete the reference", "podName", pod.Name, "targetPath", jfsSetting.TargetPath)
+					if err := resource.DelPodAnnotation(ctx, p.K8sClient, &po, []string{k}); err != nil {
+						return "", err
+					}
+				}
+			}
+			continue
+		}
 		if pod.DeletionTimestamp != nil || resource.IsPodComplete(&pod) {
 			continue
 		}
 		if pod.Spec.NodeName == jfsConfig.NodeName || pod.Spec.NodeSelector["kubernetes.io/hostname"] == jfsConfig.NodeName {
```

Comment:
check hash value?

Reply:
checked in https://github.com/juicedata/juicefs-csi-driver/blob/7096422f42978608185ea8fb87cdce34c56113e3/pkg/juicefs/mount/pod_mount.go#L327

and continue here if hash value is not equal https://github.com/juicedata/juicefs-csi-driver/blob/7096422f42978608185ea8fb87cdce34c56113e3/pkg/juicefs/mount/pod_mount.go#L336